### DISCUSSION TITLE
Supporting Prettier-specific keys in .editorconfig

### DISCUSSION THREAD

**Discussion by Drakmyth** [prettier/prettier#15404] [Ideas]
Bringing this discussion over here from https://github.com/josephfrazier/editorconfig-to-prettier/issues/24 now that https://github.com/prettier/prettier/pull/15394 has been merged:

> Hi. I'm a newcomer to both Prettier and EditorConfig but am liking what I'm seeing with both of them. I just wanted to throw another vote out for being able to support keys not defined in the EC spec from the .editorconfig file. I saw (https://github.com/josephfrazier/editorconfig-to-prettier/issues/9) and (https://github.com/josephfrazier/editorconfig-to-prettier/issues/21) were already closed on this topic, but I wanted to provide some additional points in support I came across and wasn't sure if an already closed ticket was an appropriate place to continue the discussion.
> 
> Reading https://github.com/editorconfig/editorconfig/issues/415 and https://github.com/editorconfig/editorconfig/issues/60#issuecomment-7575712 it looks as though the expectation from the EditorConfig team is that tools would be able to read domain-specific or even tool-specific properties from the .editorconfig file that aren't necessarily defined by the EditorConfig spec.
> 
> It also looks like [ReSharper](https://www.jetbrains.com/help/resharper/Using_EditorConfig.html) (and other JetBrains tools) and [Roslyn](https://learn.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2022) both heavily support custom keys for configuration.
> 
> My use case is currently very small - the only non-standard key I wish to configure right now is the bracketSameLine option - but it's annoying that I either need to have two separate config files (.editorconfig and .prettierrc) or have to move away from EditorConfig entirely to do so.
> 
> Not sure if this moves the needle on adding support for this or not. I'm nearly always on the train of not adding custom extensions for things so while my initial thought was the same as expressed in (https://github.com/josephfrazier/editorconfig-to-prettier/issues/9), seeing that the EditorConfig team is actually encouraging the practice changed my mind and seeing these issues I wanted to make sure others were aware too.

It would be convenient if prettier-specific fields were configurable via the .editorconfig file, probably with a `prettier_` prefix in the same style as the Resharper and Roslyn examples above. I wouldn't necessarily expect the same full functionality as available via a .prettierrc file (per-file overrides, etc.), but just being able to set project-global settings without needing a separate config in addition to the .editorconfig I already have would be nice. For my specific case for example: `prettier_bracket_same_line=true`.

**Comment by sosukesuzuki** [comment]
I don't use editorconfig, so I don't know how much this feature contributes to usability. And, I don't want to increase Prettier's configuration features any further...
